File: SwiggySpringBoot/swiggyClone/src/main/java/com/swiggy/swiggyClone/dataModel/RestaurantDetailResponse.java
package com.swiggy.swiggyClone.dataModel;

import java.util.List;

public class RestaurantDetailResponse extends StatusCodeModel {
    public RestaurantDetailResponse(String status, int code, String message) {
        super(status, code, message);
    }

    public RestaurantDetailResponse(String status, int code) {
        super(status, code);
    }

    private RestaurantDetailTable data;

    public RestaurantDetailResponse(String status, int code, String message, RestaurantDetailTable data) {
        super(status, code, message);
        this.data = data;
    }

    public RestaurantDetailResponse(String status, int code, RestaurantDetailTable data) {
        super(status, code);
        this.data = data;
    }

    public RestaurantDetailTable getData() {
        return data;
    }

    public void setData(RestaurantDetailTable data) {
        this.data = data;
    }
}

File: app/src/main/java/com/kotlinapp/swiggyclone/productDetail/model/RestaurantDetailResponse.kt
package com.kotlinapp.swiggyclone.productDetail.model

import com.google.gson.annotations.SerializedName

data class RestaurantDetailResponse(  @SerializedName("status"  ) var status  : String? = null,
                                      @SerializedName("code"    ) var code    : Int?    = null,
                                      @SerializedName("message" ) var message : String? = null,
                                      @SerializedName("data"    ) var data    : Data?   = Data())
data class Data (

    @SerializedName("restaurantDetailId" ) var restaurantDetailId : Int?    = null,
    @SerializedName("cuisines"           ) var cuisines           : String? = null,
    @SerializedName("location"           ) var location           : String? = null,
    @SerializedName("lat"                ) var lat                : Double? = null,
    @SerializedName("lng"                ) var lng                : Double? = null,
    @SerializedName("rating"             ) var rating             : Double?    = null,
    @SerializedName("time"               ) var time               : String? = null,
    @SerializedName("price"              ) var price              : String? = null,
    @SerializedName("restaurantName"     ) var restaurantName     : String? = null,

)